### ISSUE TITLE
UnsupportedResource Exception when running 'generate'

### ISSUE THREAD

**Issue by colinnuk** [prom3theu5/aspirational-manifests#159]
### 🔥 Bug Description
I was trying to get my Aspire project working with aspirate today. Using the Aspire preview 6 & aspirate 0.6.2-preview. But I'm encountering the below error (console log dump below).

```
── Handle Value and Parameter Substitution ────────────────────────────────────────────────────────────────────────────────────────────────
Error executing action 'SubstituteValuesAspireManifestAction':
System.NotSupportedException: Runtime type 'Aspirate.Shared.Models.AspireManifests.Components.UnsupportedResource' is not supported by
polymorphic type 'Aspirate.Shared.Models.AspireManifests.Resource'. Path: $.
     System.NotSupportedException: Runtime type 'Aspirate.Shared.Models.AspireManifests.Components.UnsupportedResource' is not supported by
     polymorphic type 'Aspirate.Shared.Models.AspireManifests.Resource'.
       at void System.Text.Json.ThrowHelper.ThrowNotSupportedException_RuntimeTypeNotSupported(Type baseType, Type runtimeType)
       at bool System.Text.Json.Serialization.Metadata.PolymorphicTypeResolver.TryGetDerivedJsonTypeInfo(Type runtimeType, out JsonTypeInfo
          jsonTypeInfo, out object typeDiscriminator)
       at JsonConverter System.Text.Json.Serialization.JsonConverter.ResolvePolymorphicConverter(object value, JsonTypeInfo jsonTypeInfo,
          JsonSerializerOptions options, ref WriteStack state)
       at bool System.Text.Json.Serialization.JsonConverter`1.TryWrite(Utf8JsonWriter writer, in T value, JsonSerializerOptions options,
          ref WriteStack state)
       at bool System.Text.Json.Serialization.Converters.DictionaryOfTKeyTValueConverter`3.OnWriteResume(Utf8JsonWriter writer, TCollection
          value, JsonSerializerOptions options, ref WriteStack state)
       at bool System.Text.Json.Serialization.JsonDictionaryConverter`3.OnTryWrite(Utf8JsonWriter writer, TDictionary dictionary,
          JsonSerializerOptions options, ref WriteStack state)
       at bool System.Text.Json.Serialization.JsonConverter`1.TryWrite(Utf8JsonWriter writer, in T value, JsonSerializerOptions options,
          ref WriteStack state)
       at bool System.Text.Json.Serialization.JsonConverter`1.WriteCore(Utf8JsonWriter writer, in T value, JsonSerializerOptions options,
          ref WriteStack state)
  at void System.Text.Json.ThrowHelper.ThrowNotSupportedException(ref WriteStack state, NotSupportedException ex)
  at bool System.Text.Json.Serialization.JsonConverter`1.WriteCore(Utf8JsonWriter writer, in T value, JsonSerializerOptions options, ref
     WriteStack state)
  at void System.Text.Json.Serialization.Metadata.JsonTypeInfo`1.Serialize(Utf8JsonWriter writer, in T rootValue, object rootValueBoxed)
  at string System.Text.Json.JsonSerializer.WriteString<TValue>(in TValue value, JsonTypeInfo<TValue> jsonTypeInfo)
  at string System.Text.Json.JsonSerializer.Serialize<TValue>(TValue value, JsonSerializerOptions options)
  at JsonNode Aspirate.Shared.Extensions.JsonExtensions.TryParseAsJsonNode<T>(T instance) in
     /home/runner/work/aspirational-manifests/aspirational-manifests/src/Aspirate.Shared/Extensions/JsonExtensions.cs:42
  at void Aspirate.Processors.Transformation.ResourceExpressionProcessor.ProcessEvaluations(Dictionary<string, Resource> resources) in
     /home/runner/work/aspirational-manifests/aspirational-manifests/src/Aspirate.Processors/Transformation/ResourceExpressionProcessor.cs:
     11
  at Task<bool> Aspirate.Commands.Actions.Manifests.SubstituteValuesAspireManifestAction.ExecuteAsync() in
     /home/runner/work/aspirational-manifests/aspirational-manifests/src/Aspirate.Commands/Actions/Manifests/
     SubstituteValuesAspireManifestAction.cs:9
  at async Task<int> Aspirate.Commands.Actions.ActionExecutor.ExecuteCommandsAsync() in
     /home/runner/work/aspirational-manifests/aspirational-manifests/src/Aspirate.Commands/Actions/ActionExecutor.cs:65
```

Further to that, once I resolve that issue, I am getting a JsonObject exception because a JsonArray is being passed in.

### 🔍 Steps to Reproduce the Bug
The apphost contains two 'unsupported resources': a postgres resource and the appInsights resource. I understand that these wont be available from aspirate - which is totally fine (they can be ignored).

Despite not selecting them as components at the prompt, the UnsupportedResource is still present in the Dictionary passed thru to this line https://github.com/prom3theu5/aspirational-manifests/blob/c7233c4a61a4ddc52253e974a2a9ede60b380035/src/Aspirate.Shared/Extensions/JsonExtensions.cs#L42

### 🧯 Possible Solution
On a local branch I filtered out the unknown resources in the caller (ResourceExpressionProcessor.ProcessEvaluations) by adding a LINQ `.Where(r => r.Value.Type != "unknown.resource.v0")`. This fixed the issue locally. Happy to push this as a fix if you think it'll make sense.
```
    public void ProcessEvaluations(Dictionary<string, Resource> resources)
    {
        resources.EnsureBindingsHavePorts();
        var jsonDocument = resources
            .Where(r => r.Value.Type != "unknown.resource.v0")
            .ToDictionary(p => p.Key, p => p.Value)
            .TryParseAsJsonNode();
        var rootNode = jsonDocument.Root;

        jsonExpressionProcessor.ResolveJsonExpressions(rootNode, rootNode);

        HandleSubstitutions(resources, rootNode);
    }
```

**Comment by colinnuk** [comment]
I was just playing around some more and the Aspire Sample works just fine.... so the issues are likely something I need to change in my project setup.

**Comment by prom3theu5** [comment]
Thanks for debugging and breaking it down

Yeah that would work - i'd prefer to use pattern matching rather than the string so `r.Value is not UnsupportedResource)`
I'll get a fix out now

**Comment by prom3theu5** [comment]
Just merged the fix into v0.6.3-preview
Again thanks for detailed breakdown
